### PR TITLE
Builtins reference: Add more rules for List/reverse

### PR DESCRIPTION
**Pull request by sjakobi**

### CODE REVIEW

**Comment by sjakobi** [comment]
Is there any interest in getting some of these standardized?

**Comment by Gabriel439** [comment]
@sjakobi: We should probably hold off on additional standard simplifications, mainly because we could in theory need to standardize O(N^2) rules specifying how N built-ins interact

File: docs/references/Built-in-types.md [hunk 1]
```diff
@@ -1143,7 +1143,17 @@ List/reverse a ([] : List a) = [] : List a
...
+
 List/reverse a (xs # ys) = List/reverse a ys # List/reverse a xs
+
+List/head a (List/reverse a xs) = List/last a xs
+
+List/last a (List/reverse a xs) = List/head a xs
+
+List/length a (List/reverse a xs) = List/length a xs
+
+List/reverse a (List/indexed a (List/reverse a xs)) = List/indexed a xs
```
Comment (Gabriel439) [inline docs/references/Built-in-types.md]: I don't think this last rule is correct.  Here is the counter-example:

```dhall
let xs = [ 2, 3, 5 ]
  
let a = Natural

in    assert
    :   List/reverse
          { index : Natural, value : a }
          (List/indexed a (List/reverse a xs))
      ≡ List/indexed a xs
```

... which fails with:

```
$ dhall --file ./test.dhall

Use "dhall --explain" for detailed errors

xs : List Natural
a : Type

Error: Assertion failed

[ - { index = 2, value = 2 }
, + { index = 0, value = 2 }
, …
, - { index = 0, value = 5 }
, + { index = 2, value = 5 }
]


5│     assert : List/reverse { index : Natural, value : a } (List/indexed a (List/reverse a xs)) === List/indexed a xs

./test.dhall:5:5
```

Comment (sjakobi) [reply docs/references/Built-in-types.md]: Thanks! Not sure what I was thinking! :)

Comment (Gabriel439) [reply docs/references/Built-in-types.md]: @sjakobi: You're welcome! 🙂

**Pull request merged**